include/ncengine/physics/PhysicsBody.h

```diff
@@ -58,6 +58,10 @@ class PhysicsBody final : public ComponentBase
         void Sleep() noexcept { m_awake = false; }
 
         auto GetProperties() const noexcept -> const PhysicsProperties& { return m_properties; }
+        auto GetLinearFreedom() const noexcept -> Vector3 { return ToVector3(GetLinearFreedomXM()); }
```

Comment:
Not exactly sure how ImGui works internally, but are these going to be called every frame? If so, consider caching the NC types.

Reply:
Its only called during serialization right now, so very rarely. My gut tells me caching for imgui would likely have no effect, and may be a pessimization for physics code due to increasing obj size. It would only be called once per frame if the currently selected object has a physics body.